Remove space after if at StandardCompiler line 227

davidcian opened the pull request:
Test CircleCI comments

Commit:
Update contributing doc to clarify tests

Fix line endings

Update docs/contributing.rst

Co-Authored-By: Bhargava Shastry <bhargava.shastry@ethereum.org>

Remove duplication

Add back URL to correct evmone version

Add Istanbul

erak committed:
Adds script which compiles docs example with minimum compiler version.

gitpusha committed:
Added note recommending to avoid relying on hardcoded gas values

cameel committed:
[yul-phaser] Add RelativeProgramSize metric

cameel committed:
[yul-phaser] Refactor the code from main into a Phaser class

cameel committed:
[yul-phaser] Extract the code that controls execution of algorithm rounds from GeneticAlgorithm into AlgorithmRunner

cameel committed:
[yul-phaser] Phaser: Refactor object creation in runAlgorithm() into a set of factories

cameel committed:
[yul-phaser] Phaser: Split command-line option description into groups

cameel committed:
[yul-phaser] Phaser: Extract preparation of option description into a separate function

cameel committed:
[yul-phaser] Phaser: Make all option descriptions consistenly end with a full stop

cameel committed:
[yul-phaser] AlgorithmRunner: Create a structure to store runner's options

cameel committed:
[yul-phaser] Add --rounds option

cameel committed:
[yul-phaser] Add --random-population option

cameel committed:
[yul-phaser] Add --population option

cameel committed:
[yul-phaser] Common: Add readLinesFromFile()

cameel committed:
[yul-phaser] Add --population-from-file option

cameel committed:
[yul-phaser] AlgorithmRunner: Population autosave

cameel committed:
[yul-phaser] Add --population-autosave option

cameel committed:
[yul-phaser] Phaser: Name option parameters in command-line help

cameel committed:
[yul-phaser] Add --metric option

cameel committed:
[yul-phaser] Add --relative-metric-scale option

cameel committed:
[yul-phaser] Add FitnessMetricCombination and derived classes for average, sum, min and max

cameel committed:
[yul-phaser] Phaser: Accepting multiple input programs

- Use average of metric values for individual programs as the overall metric.

bshastry committed:
Add a security policy and link it in README.

Co-Authored-By: chriseth <chris@ethereum.org>

cameel committed:
[yul-phaser] Phaser: Reimplement << and >> operators using serializeChoice() and deserializeChoice()

cameel committed:
[yul-phaser] Add --metric-aggregator option

cameel committed:
[yul-phaser] Add ProgramCache class

cameel committed:
[yul-phaser] Make FitnessMetric::evaluate() non-const

cameel committed:
[yul-phaser] Extract ProgramBasedMetric base class from ProgramSize

cameel committed:
[yul-phaser] ProgramBasedMetric: Add the ability to use ProgramCache

cameel committed:
[yul-phaser] Add --program-cache option

cameel committed:
[yul-phaser] Add --min-chromosome-length and --max-chromosome-length options

cameel committed:
[yul-phaser] Add options for all algorithm-specific parameters

cameel committed:
[yul-phaser] Phaser: Change default metric to RelativeCodeSize

cameel committed:
[yul-phaser] AlgorithmRunner: Duplicate chromosome randomisation

cameel committed:
[yul-phaser] AlgorithmRunner: Count rounds from 1

cameel committed:
[yul-phaser] AlgorithmRunner: Add support for ProgramCache

cameel committed:
[yul-phaser] Add --show-initial-population option

cameel committed:
[yul-phaser] Add --show-only-top-chromosome and --hide-round options

cameel committed:
[yul-phaser] AlgorithmRunner: Print elapsed time after each round

cameel committed:
[yul-phaser] Program: Output operator for ErrorList

- It does not seem universal enough to put it in liblangutil but in the scope of yul-phaser it's generic enough so I'm going to keep it together with Program.

cameel committed:
[yul-phaser] main: Dealing with uncaught exceptions

cameel committed:
[yul-phaser] Returning an ErrorList from Program::load() if program has errors and printing them in Phaser

cameel committed:
[yul-phaser] Move handling of boost::program_options::error to the top-level exception handler

cameel committed:
[yul-phaser] Create BadInput exception hierarchy and make exceptions used by Phaser more specific

cameel committed:
[yul-phaser] Phaser: Returning exit code 0 instead of 2 when --help is requested

- I don't think we really need to discern it from normal operation. It's what user requested so it's a success.

cameel committed:
[yul-phaser] Don't return exit code from Phaser::main() and just assume 0 if it does not throw.

bshastry committed:
solc fuzzers: Use compiler stack for fuzzing

cameel committed:
[yul-phaser] Add ProgramCacheFactory class

cameel committed:
[yul-phaser] Add --mode option

cameel committed:
[yul-phaser] Add print-optimised-programs and print-optimised-asts modes

cameel committed:
[yul-phaser] ProgramCache: Add ability to gather cache stats

cameel committed:
[yul-phaser] Add --show-cache-stats option

cameel committed:
[yul-phaser] Add --chromosome-repetitions option

cameel committed:
[yul-phaser] main: Add missing underscores to parameter names

cameel committed:
[yul-phaser] Rename Common to Helpers in tests (file move)

- I want to add Common.h in the non-test code and it's getting confusing. I think it's better to have them named differently.

cameel committed:
[yul-phaser] Rename Common to Helpers in tests (update references)

cameel committed:
[yul-phaser] Add Common module

cameel committed:
[yul-phaser] Add serializeChoice() and deserializeChoice()

Commit:
Zero initialize memory arrays

cameel committed:
[yul-phaser] Add --no-randomise-duplicates option

cameel committed:
[yul-phaser] TestHelpers: Add TemporaryDirectory class

cameel committed:
[yul-phaser] ProgramSize metric: Refactor repeating parts of test code

- Also replace test fixture constructor with direct member initialisation

cameel committed:
[yul-phaser] More data accessors in metric and algorithm classes

cameel committed:
[yul-phaser] Phaser: Tests for factories

cameel committed:
[yul-phaser] Program: Explicitly return variants values

- Otherwise the code fails to build with on GCC 7.4.0 on Ubuntu in the CI pipeline

a3d4 committed:
Introduced TestCaseReader.

a3d4 committed:
Decoupled error checking and report printing in CommonSyntaxTest.

cameel committed:
[yul-phaser] AlgorithmRunner: A stronger test for run() output

cameel committed:
[yul-phaser] AlgorithmRunner: Make all tests use population from AlgorithmRunnerFixture

a3d4 committed:
Unified dialect selection in libyul/SyntaxTest and YulOptimizerTest.

Marenz committed:
Disallow private functions from being overridden

Marenz committed:
Add private-can-be-overridden bug to buglist

cameel committed:
[yul-phaser] AlgorithmRunner: Measure CPU time rather than wall-clock time

cameel committed:
[yul-phaser] AlgorithmRunner: Print total time when showing only the top chromosome

cameel committed:
[yul-phaser] Population: Print individuals in a more compact way

cameel committed:
[yul-phaser] Add --show-seed option and don't print seed by default

aarlt committed:
Extracted 292 tests from SolidityEndToEndTest.cpp

ekpyron committed:
Legacy codegeneration for immutable state variables.

Marenz committed:
Validate immutable variables

aarlt committed:
Add --show-metadata to enable metadata output.

christianparpart committed:
Add antlr-based grammar and test it.

cameel committed:
[yul-phaser] Program: Add parseObject()

cameel committed:
[yul-phaser] Program: Switch from using parseCode() to parseObject()

ekpyron committed:
Tests.

ekpyron committed:
Anything outside storage is always a pointer.

davidcian committed:
Make CircleCI post diff review for style violation

The CircleCI robot now also posts a review comment
directly on the diff (on GitHub, the changes tab),
where the style violations take place.

chriseth committed:
Merge pull request #8481 from aarlt/add-show-metadata

Add --show-metadata to enable metadata output.

chriseth committed:
Merge pull request #8457 from ethereum/add-to-bug-list

Add private-can-be-overridden bug to buglist

chriseth committed:
Merge pull request #8421 from imapp-pl/yul-phaser-refactor-main

[yul-phaser] Refactoring in main

Marenz committed:
Fix broken yul-object link in docs

aarlt committed:
Add tools: create_traces.sh, remove-testcases.py & verify-testcases.py

chriseth committed:
Merge pull request #8496 from ethereum/fixdoclinks

Fix broken yul-object link in docs

chriseth committed:
Merge pull request #8290 from gitpusha/docs-update-staticcall-gascounting

added note about gas counting and its effect on staticall.gas(gasLimi…

chriseth committed:
Merge pull request #8455 from ethereum/private-inherit

Prevent private functions from being overridden

chriseth committed:
Set version to 0.5.17.

chriseth committed:
Merge tag 'v0.5.17' into develop

a3d4 committed:
Moved "step" from settings to expectations.

chriseth committed:
Merge pull request #8422 from imapp-pl/yul-phaser-error-handling

[yul-phaser] Error handling

chriseth committed:
Merge pull request #8442 from a3d4/fix-8244-refactor-testcase-classes

Refactor TestCase classes

chriseth committed:
Merge pull request #8423 from imapp-pl/yul-phaser-population-and-algorithm-options

[yul-phaser] Population and algorithm options

chriseth committed:
Merge pull request #8502 from mijovic/testDocsPragma

[docs] Compile examples with minimal compiler version

a3d4 committed:
Unified use of settings. Removed a couple of unused functions.

chriseth committed:
Sort tests.

chriseth committed:
Merge pull request #8464 from aarlt/end-to-end-verified

[test] Extraction of 292 tests from SolidityEndToEndTest.cpp

chriseth committed:
Merge pull request #8509 from ethereum/sortSemanticTests

Sort tests.

chriseth committed:
Merge pull request #8449 from imapp-pl/yul-phaser-multi-program-support

[yul-phaser] Multi-program support

chriseth committed:
Documentation for immutables.

chriseth committed:
Merge pull request #8508 from a3d4/again-refactor-testcase-classes

More refactoring of TestCase classes

cameel committed:
[yul-phaser] Add --prefix option

Commit:
Fix Yul codegen when dynamic array is used as rhs of assignment

aarlt committed:
Extract 45 tests from SolidityEndToEndTest.cpp

iamdefinitelyahuman committed:
fix typo in docstring error message

davidcian commented:
There was an error when running `chk_coding_style` for commit `e15cf4b6fc11b9e5daa450ce637eaaa65f4eafb6`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `775f1ceabaa576a080ee571267f46f5efdaaf63a`:
```
Error: Trailing whitespace found:
 scripts/report_errors.sh:49: --data "{\"body\": \"There was an error when running \`
CIRCLE_JOB\` for commit \`
CIRCLE_SHA1\`:\n\`\`\`\n
FORMATTED_ERROR_MSG\n\`\`\`\nPlease check that your changes are working as intended.\"}" 

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `967225aa7d3a204214406f737f5a0e942ccd7aa5`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `38ec7bd52dc981c1769964e7e561c4f4d22e632f`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `1e30625e673e42be37d967856d48a66aca88f318`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `aa60deb6b32bdcb978c97b9b1c630ec32ef6331b`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented on libsolidity/interface/StandardCompiler.cpp:
```diff
@@ -224,7 +224,7 @@ bool isArtifactRequested(Json::Value const& _outputSelection, string const& _fil
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

a3d4 committed:
Improved `.clang_format` file.

chriseth committed:
Merge pull request #8520 from a3d4/improve-clang-format

Improved .clang_format file

chriseth committed:
Merge pull request #8512 from ethereum/yul_rhs_array

Fix Yul codegen when dynamic array is used as rhs of assignment

mijovic committed:
[Sol2Yul] Adding support for constructor with parameters

chriseth committed:
Merge pull request #8491 from ethereum/publicStateVarOverrideDynamicMemory

Always mark memory reference types as pointers.

chriseth committed:
Merge pull request #8519 from iamdefinitelyahuman/develop

fix typo in docstring error message

a3d4 committed:
Fix #8427: Promoted typeError to fatalTypeError in ReferencesResolver::endVisit(UserDefinedTypeName).

chriseth committed:
Merge pull request #8505 from aarlt/antlr-grammar-aarlt

Antlr4 based grammar, with CI integration.

Commit:
Add natspec test with too many return tags

MrChico committed:
CommandLineInterface: add storage-layout option to --combined-json

chriseth committed:
Merge pull request #8524 from ethereum/test_natspec_return

Add natspec test with too many return tags

chriseth committed:
Merge pull request #8400 from ethereum/fix-8395

[solc] Standard compiler interface to permit exception leaks and use it in fuzzer

MrChico committed:
CMakeLists: ensure OSX deployment target supports std::visit

chriseth committed:
Merge pull request #8513 from imapp-pl/yul-phaser-parsing-yul-objects

[yul-phaser] Parsing Yul objects

chriseth committed:
Merge pull request #8514 from imapp-pl/yul-phaser-prefix-option

[yul-phaser] --prefix option

chriseth committed:
Refactor: Replace inheritance hierarchy by most derived contract.

Commit:
Symbolic state

Commit:
[SMTChecker] Add SortProvider

chriseth committed:
Merge pull request #8472 from ethereum/immutableCodegen

Immutable codegen

chriseth committed:
Merge pull request #8451 from imapp-pl/yul-phaser-program-cache

[yul-phaser] Program cache

chriseth committed:
Merge pull request #8522 from a3d4/fix-8427-ast-types-compiler-error

Promoted typeError to fatalTypeError in ReferencesResolver

chriseth committed:
Merge pull request #8452 from imapp-pl/yul-phaser-more-output

[yul-phaser] More output

randomnetcat committed:
Convert operator+(Population, Population) into a hidden friend

chriseth committed:
Merge pull request #8390 from ethereum/fix-8389

SECURITY.md: security policy for Solidity

Commit:
Merge pull request #8530 from MrChico/oldosxbuild

Oldosxbuild

Commit:
Merge pull request #8531 from ethereum/smt_sort_provider

[SMTChecker] Add SortProvider

chriseth committed:
Move files.

chriseth committed:
Merge pull request #8518 from aarlt/extract-more-testcases

[test] Extract 45 more tests from SolidityEndToEndTest.cpp

chriseth committed:
Merge pull request #8533 from ethereum/refactorVirtualResolution

Refactor virtual resolution

chriseth committed:
Merge pull request #8529 from MrChico/develop

CommandLineInterface: add storage-layout option to --combined-json

randomnetcat committed:
Delete copy/move on annotation polymorphic bases

randomnetcat committed:
Remove forward declaration of class Population

mijovic committed:
Enabling pragma with > in minimum version check

christianparpart committed:
Removing grammar.txt in favor of Solidity.g4 (ANTLR based grammar)

a3d4 committed:
Fix #8450. Prevented internal compiler errors when assigning nested tuples.

chriseth committed:
Merge pull request #8549 from ethereum/remove-grammar-txt

Removing grammar.txt in favor of Solidity.g4 (ANTLR based grammar)

chriseth committed:
Merge pull request #8539 from random-internet-cat/population-op-plus

Convert operator+(Population, Population) into a hidden friend

chriseth committed:
Merge pull request #8534 from mijovic/sol2YulConstructorParams

[Sol2Yul] Adding support for constructor with parameters

aarlt committed:
Disallow access to functions from inline assembly.

chriseth commented:
This looks already really nice!
If I read https://developer.github.com/v3/pulls/comments/#create-a-comment right, then you need the header `application/vnd.github.comfort-fade-preview+json` for the "line" option to work. Did you set that?

chriseth committed:
Merge pull request #8550 from a3d4/fix-8450-typechecker-compiler-error

Fix  a couple of internal compiler errors when assigning nested tuples

chriseth committed:
Merge pull request #8548 from mijovic/testDocsPragma

Enabling pragma with > in minimum version check

aarlt committed:
Apply modernize-use-override.

aarlt committed:
Apply modernize-use-nullptr.

aarlt committed:
Apply modernize-use-emplace.

chriseth committed:
Merge pull request #8553 from aarlt/inline_asm_function_access

Disallow function access in inline assembly

chriseth committed:
Merge pull request #8510 from ethereum/immutableDocs

Documentation and grammar for immutables.

chriseth committed:
Fix call to internalDispatch.

Marenz committed:
Always return `this` for `EventDefinition::resolveVirtual`

Instead of asserting.

chriseth committed:
Merge pull request #8563 from ethereum/fixInternalDispatch

Fix call to internalDispatch.

ekpyron committed:
Restrict size for dynamic memory array creation.

ekpyron committed:
Add buglist entry.

mijovic committed:
[docs] Renaming CI job for documentation pragma version check

chriseth committed:
Merge pull request #8566 from mijovic/testDocsPragma

[docs] Renaming CI job for documentation pragma version check

chriseth committed:
Merge pull request #8565 from ethereum/memoryCreationOverflowBug

Memory creation overflow bug.

aarlt committed:
[build-system] Add -Wpessimizing-move & -Wredundant-move warnings.

-Wpessimizing-move warns when a call to std::move would prevent copy elision
if the argument was not wrapped in a call.  This happens when moving a local
variable in a return statement when the variable is the same type as the
return type or using a move to create a new object from a temporary object.

-Wredundant-move warns when an implicit move would already be made, so the
std::move call is not needed, such as when moving a local variable in a return
that is different from the return type.

davidcian commented:
There was an error when running `chk_coding_style` for commit `e5d754f87104860d0c5be8bc14976945ff1ff391`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

```
Please check that your changes are working as intended.

davidcian commented on libsolidity/interface/StandardCompiler.cpp:
```diff
@@ -224,7 +224,7 @@ bool isArtifactRequested(Json::Value const& _outputSelection, string const& _fil
 /// @returns true if any binary was requested, i.e. we actually have to perform compilation.
 bool isBinaryRequested(Json::Value const& _outputSelection)
 {
-	if (!_outputSelection.isObject())
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())

davidcian replied:
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

davidcian replied:
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

davidcian replied:
xxx

davidcian replied:
Coding style error

davidcian commented:
> This looks already really nice!
> If I read https://developer.github.com/v3/pulls/comments/#create-a-comment right, then you need the header application/vnd.github.comfort-fade-preview+json for the "line" option to work. Did you set that?

Yep, no idea how I missed that 🤦‍♂. It works like a charm now. I'll just clean the code up a bit and then submit the PR.

ekpyron committed:
Properly handle assignments of immutables at declaration.

hrkrshnn committed:
Removed redundant declaration check; changed relevant test cases

chriseth committed:
Merge pull request #8574 from ethereum/immutableAssignAtDecl

Properly handle assignments of immutables at declaration.

chriseth committed:
Merge pull request #8569 from aarlt/clang-tidy-apply-modernize-use-nullptr

clang-tidy: Apply modernize-use-nullptr.

chriseth commented:
Oh that's wonderful, thanks a lot!

hrkrshnn committed:
Extended test-coverage for storage declaration

hrkrshnn committed:
New folder localStorageVariable for new testcases; added two new tests

chriseth committed:
Merge pull request #8567 from ethereum/storage-pointer-checker

Removed redundant storage declaration check; test coverages

hrkrshnn committed:
Added changelog

chriseth committed:
Merge pull request #8576 from ethereum/add-changelog-storage-pointer

Added changelog

chriseth committed:
Merge pull request #8475 from ethereum/immutable-functioncallgraph

implement ImmutableValidator class

chriseth committed:
Merge pull request #8462 from ethereum/allocate_memory_zero_function

[Yul codegen] Zero initialize memory arrays

hrkrshnn committed:
Added error for interface function with modifiers; test case

chriseth committed:
Accessors for immutable variables.

ekpyron committed:
Debug information for immutable references.

chriseth committed:
Changelog entry for immutable variables.

chriseth committed:
Merge pull request #8578 from ethereum/interface-function-modifier-error

Added error for interface functions that have modifiers; test case

chriseth committed:
Skip verification for external access.

chriseth committed:
Merge pull request #8580 from ethereum/publicImmutables

Accessors for immutable variables.

davidcian commented:
There was an error when running `chk_coding_style` for commit `870f26eed0bad7b99b24702371387024c5991b7a`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `a338cda683a14e3ce054de39697794bead2f9bf5`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `66599f441266381e04857225799311e5ad1f04ab`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `4ecebe340c43bc11548e921b35584d619e593b6b`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `4ecebe340c43bc11548e921b35584d619e593b6b`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `41ece36feceaaaaf1f3fdfbfd9525722449b4e63`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `0f55d113c1e56586c03cdb95963daa3d07ed7fb6`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `3fb9a18d00535d6f17000ebd33ac9398a4392382`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `486c03ace0e5716970c7fde2efaa7b6a987ce87f`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `fa63843e706bca32fbc8e2f1f89ff11216359cab`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `b58bab07c0814884ec634a0c5ab6b3c31ab85b79`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented:
There was an error when running `chk_coding_style` for commit `9d810a55d7bb6241986187410df958bed9ea7ab3`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

davidcian commented on libsolidity/interface/StandardCompiler.cpp:
```diff
@@ -257,7 +257,7 @@ bool isEwasmRequested(Json::Value const& _outputSelection)
 	for (auto const& fileRequests: _outputSelection)
 		for (auto const& requests: fileRequests)
 			for (auto const& request: requests)
-				if (request == "ewasm" || request == "ewasm.wast")
```
xxx

davidcian replied:
Coding style error

davidcian commented:
There was an error when running `chk_coding_style` for commit `5d64bd1d049aaf2702b17c6ee782eba0db325af7`:
```
Coding style error:
 libsolidity/interface/StandardCompiler.cpp:227: if(!_outputSelection.isObject())
 libsolidity/interface/StandardCompiler.cpp:260: if(request == "ewasm" || request == "ewasm.wast")

```
Please check that your changes are working as intended.

ekpyron committed:
Merge pull request #8584 from random-internet-cat/annotation-constructors

Delete copy/move on annotation polymorphic bases to improve safety

ekpyron committed:
Disallow external function pointers as immutables.

Commit:
Merge pull request #8582 from ethereum/chriseth-patch-1

Changelog entry for immutable variables.

ekpyron committed:
Merge pull request #8573 from ethereum/add-move-specific-warnings

[build-system] Add -Wpessimizing-move & -Wredundant-move warnings.

randomnetcat committed:
Delete copy and move operations for ASTVisitor and ASTConstVisitor

a3d4 committed:
Fix tuple assignments with multi-slot components.

cameel committed:
reference-types.rst: Minor text correction, itself -> themselves

cameel committed:
reference-types.rst: Fix incorrectly wrapped line in the array slice section

chriseth committed:
Merge pull request #8609 from ethereum/trivial-doc-fixes-in-reference-types-rst

Trivial documentation fixes in reference-types.rst

Commit:
Allow constructing symbolic arrays from smt sort

ekpyron committed:
Merge pull request #8586 from ethereum/immutableExternalFunctionPointers

Disallow external function pointers as immutables.

chriseth committed:
Prepare changelog for 0.6.5.

ekpyron committed:
Format immutable references similarly to link references.

ekpyron committed:
Document immutable references output.

Commit:
Merge pull request #8611 from ethereum/smt_array_variable_sort

[SMTChecker] Allow constructing symbolic arrays from smt sort

chriseth committed:
Update buglist.

chriseth committed:
Merge pull request #8613 from ethereum/prepareNextRelease

Prepare changelog for 0.6.5.

chriseth committed:
Merge pull request #8592 from random-internet-cat/ast-visitor-constructor

Delete copy and move operations for ASTVisitor and ASTConstVisitor

chriseth committed:
Merge pull request #8581 from ethereum/immutableDebug

Debug information for immutable references.

Marenz committed:
Allow for per-parameter literalValues builtin functions

chriseth committed:
Merge pull request #8616 from ethereum/develop

Merge develop into release for 0.6.5

Commit:
Merge pull request #8579 from ethereum/smt_symbolic_state

[SMTChecker] Symbolic state

chriseth committed:
Merge pull request #8568 from aarlt/clang-tidy-apply-modernize-use-override

clang-tidy: Apply modernize-use-override.

randomnetcat committed:
Use readlink -f instead of realpath in ASTImportTest.sh

mijovic committed:
Refactoring nextConstructor by moving it from CompilerContext to ContractDefinition

chriseth committed:
Set version to 0.6.6.

chriseth committed:
Merge pull request #8620 from ethereum/setVersionTo066

Set version to 0.6.6.

chriseth committed:
Merge pull request #8621 from ethereum/release

Merge release back into develop.

chriseth committed:
Merge pull request #8617 from random-internet-cat/realpath-fix

Use readlink -f instead of realpath in ASTImportTest.sh

chriseth committed:
Merge pull request #8618 from mijovic/nextConstructorRefactor

Refactoring nextConstructor by moving it from CompilerContext to ContractDefinition

Commit:
[SMTChecker] Fix ICE in CHC internal calls

Commit:
Merge pull request #8624 from ethereum/smt_fix_internal_call_chc

[SMTChecker] Fix ICE in CHC internal calls

chriseth committed:
Export immutability.

aarlt committed:
Fix ownership.

chriseth committed:
Merge pull request #8570 from aarlt/clang-tidy-apply-modernize-use-emplace

clang-tidy: Apply modernize-use-emplace.

hrkrshnn committed:
TypeChecker error when encoding functions with call options; tests

hrkrshnn committed:
Added changelog

chriseth committed:
Merge pull request #8628 from ethereum/fiximmutableastexport

Export immutability.

chriseth committed:
Merge pull request #8629 from ethereum/fix_ownership

Fix ownership.

chriseth committed:
Merge pull request #8607 from a3d4/fix-nested-tuples

Fixed nested tuples on the right-hand side.

aarlt committed:
Fix ownership, part two.

chriseth committed:
Merge pull request #8630 from ethereum/encoding-bug-functionCallOption

TypeChecker error when encoding functions with call options; tests

Satoshi-Kusumoto committed:
fix link

chriseth committed:
Merge pull request #8638 from Satoshi-Kusumoto/patch-1

fix doc link

chriseth committed:
Merge pull request #8637 from ethereum/fix_ownership2

Fix ownership part two.

a3d4 committed:
Removed a redundant else clause.

chriseth committed:
Prepare 0.6.6

chriseth committed:
Merge pull request #8641 from a3d4/refactor-redundant-else

Minor refactoring. Removed a redundant else clause

chriseth committed:
Merge pull request #8643 from ethereum/prepare066

Prepare 0.6.6

chriseth committed:
Merge pull request #7632 from ethereum/fix/tests-docs

[DOCS] Clarify testing section

chriseth committed:
Merge pull request #8626 from ethereum/immutable-functioncallgraph

Prepare literalArguments for immutable builtin functions
